src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -97,9 +131,34 @@ public CommandResult addUserTags(Model model) throws CommandException {
         model.getObservablePersonList().forEach(person ->
                 person.setCommonModules(editedUser.getImmutableModuleTags()));
 
+        for (ModuleTag tag : moduleTags) {
+            String day = tag.getDay();
+            String start = tag.getStartTime();
+            String end = tag.getEndTime();
+            if (day == null || start == null || end == null) {
```

Comment:
It would be better to use `Optional.ofNullable()` but if you are still uncomfortable with using Monads, it is fine.

Reply:
Or alternatively, use requireAllNonNull.